core/io/dir_access.cpp

```diff
@@ -339,6 +339,8 @@ String DirAccess::get_full_path(const String &p_path, AccessType p_access) {
 }
 
 Error DirAccess::copy(const String &p_from, const String &p_to, int p_chmod_flags) {
+	ERR_FAIL_COND_V_MSG(p_from == p_to, ERR_INVALID_PARAMETER, "Source and destination path are equal.");
```

Comment:
Should this be an error or just return `OK`? I'd suggest the latter as it's not necessarily invalid as such, just a no-op

Reply:
Also this will not catch some cases, this should still happen if you do:
```gdscript
DirAccess.copy("./foo.txt", "foo.txt")
```
Or other non-simplified paths

Reply:
Well we don't have functions to resolve relative paths. You could even do `../SameFolder/file` and it's difficult to determine the absolute target file.
I just wanted to handle the simple case, which might be most common (in my case I tried recursively copying files and used the same root directory by mistake).

Reply:
I think there's a danger of doing a partial check, and I don't think it's the most common case, and it would be the case where it's not trivial to do the checking yourself

If people think they can rely on a check they might become careless and get in trouble when it doesn't work

Reply:
> If people think they can rely on a check they might become careless

They have no reason to think that. They will only discover the safeguard when they manage to activate it. I only opened the PR because it happened to me.

Reply:
You mean they won't go "oh, it'll stop it for me" if they see an error message? People tend to become reliant on error checks in my experience